### PR TITLE
replace dist_skel with discrete_pmf function

### PR DESCRIPTION
**Pull request by sbfnk**
## Description

This PR reflects an internal change, replacing the `dist_skel()` function with a `discrete_pmf()` function that reflects the core functionality needed here (returning the pmf of a discretised continuous distribution).

It implements a `tolerance` argument and thus paves the way towards addressing #533.

## Initial submission checklist

- [ ] ~My PR is based on a package issue and I have explicitly linked it.~ Internal non-consequential change.
- [X] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [X] I have added or updated unit tests where necessary.
- [X] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [X] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [ ] ~I have added a news item linked to this PR.~ Internal change only (no functionality change) and thus nothing to report to users.

## After the initial Pull Request 

- [ ] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: R/dist_spec.R [hunk 1]
```diff
@@ -34,151 +22,67 @@
...
-    } else {
-      rdist(n)
-    }
+  ## determine pmf
+  max_value <- ceiling(max_value)
+  if (max_value < width) {
+    cmf <- c(0, 1)
+  } else {
+    x <- seq(width, max_value, by = width)
+    cmf <- c(0, updist(width), (updist(x) + updist(x + width))) /
```
Comment (seabbs) [inline R/dist_spec.R:79]: docs should probably mention what kind of discretisation this is and if its an approximation (which I think it is?)

Comment (sbfnk) [reply R/dist_spec.R:79]: you mean adding to this?
https://github.com/epiforecasts/EpiNow2/blob/e95da199cfd77b79daca81e369fdbef33bc67bc9/R/dist_spec.R#L8

or somewhere else?

Comment (sbfnk) [reply R/dist_spec.R:92]: Addressed in 0a0538ca523a1f403a534a0e4f49a1ca69aa145f

File: R/dist_spec.R [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
-#'
-#' @param discrete Logical,  defaults to `FALSE`. Should the probability
-#'   distribution be discretised. In this case each entry of the probability
-#'   mass function corresponds to the 2-length interval ending at the entry
-#'   except for the first interval that covers (0, 1).  That is, the probability
-#'   mass function is a vector where the first entry corresponds to the integral
-#'   over the (0,1] interval of the continuous distribution, the second entry
-#'   corresponds to the (0,2] interval, the third entry corresponds to the (1,
-#'   3] interval etc.
+#' This function returns the probability mass function of a discretsied and
```
Comment (jamesmbaazam) [inline R/dist_spec.R:4]:
```suggestion
#' This function returns the probability mass function of a discretised and
```

File: R/dist_spec.R [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
+#' parameters.
+#'
+#' # Methodological details
+#'
+#' The probability mass function of the discretised probability distribution is
+#'   a vector where the first entry corresponds to the integral over the (0,1]
+#'   interval of the corresponding continuous distribution (probability of
+#'   integer 0), the second entry corresponds to the (0,2] interval (probability
+#'   mass of integer 1), the third entry corresponds to the (1, 3] interval
+#'   (probability mass of integer 2) etc. This approximates the true probability
```
Comment (jamesmbaazam) [inline R/dist_spec.R:15]:
```suggestion
#'   (probability mass of integer 2), etc. This approximates the true probability
```

**Comment by sbfnk** [comment]
Happy for it to be merged (need an approving review from someone)?

**Review by jamesmbaazam — APPROVED** [review]

**Pull request merged**